src/private/authnPrivateHandler.ts

```diff
@@ -0,0 +1,60 @@
+import { jsonCodec, PrivateNatsHandler } from "../nats/nats";
+import { headers, Subscription } from "nats";
+import { HANDLERS_SUBJECTS } from "../config";
+import { AuthnInterface, AuthnResponseInterface } from "../interfaces/authn";
+import AuthnCommand from "../commands/authn/authnCommand";
+import CreateSessionCommand from "../commands/session/createSessionCommand";
+import { CreateSessionResponse } from "../interfaces/session";
+import { authnService, sessionService } from "../di.config";
+
+export const authnPrivateHandlers: PrivateNatsHandler[] = [
+    [
+        HANDLERS_SUBJECTS.AUTHN,
+        async (subscription: Subscription): Promise<void> => {
+            for await (const message of subscription) {
+                try {
+                    const data = jsonCodec.decode(
+                        message.data
+                    ) as AuthnInterface;
```

Comment:
we don't need to typecast the payload here. We directly pass it to `AuthnCommand()` and it will verify the shape of the payload and then assign it a type for us. I think AuthInterface could be removed across the code base